lib/import/publishing_api_importer.rb

```diff
@@ -0,0 +1,100 @@
+module Import
+  class PublishingApiImporter
+
+    def initialize(checker_db, progress_reporter, publishing_api_url)
+      @checker_db = checker_db
+      @thread_pool = Thread.pool(1)
+      @publishing_api_url = publishing_api_url
+      @progress_reporter = progress_reporter
+    end
+
+    def import
+      create_publishing_api_tables
+
+      import_publishing_api_batches do |batch_data|
+        @thread_pool.process {
+          import_content(batch_data.map { |content_id_data| Import::PublishingApiDataPresenter.present_content(content_id_data) })
+          import_content_links(batch_data.flat_map { |content_id_data| Import::PublishingApiDataPresenter.present_links(content_id_data) })
+        }
+      end
+      @progress_reporter.message('publishing api import', "waiting for #{@thread_pool.backlog} remaining tasks to complete")
+      @thread_pool.shutdown
+      @progress_reporter.message('publishing api import', 'finished')
+    end
+
+  private
+
+    BATCH_SIZE = 1000
+    NIL_UUID = '00000000-0000-0000-0000-000000000000'
+
+    def create_publishing_api_tables
+      @checker_db.create_table(
+          table_name: 'publishing_api_content',
+          columns: [
+              'content_id text',
+              'publishing_app text',
+              'format text',
+              'ever_published text',
+          ],
+          index: ['content_id'],
+      )
+
+      @checker_db.create_table(
+          table_name: 'publishing_api_link',
+          columns: [
+              'content_id text',
+              'link_type text',
+              'link_content_id text',
+          ],
+          index: ['content_id'],
+      )
+    end
+
+    def import_publishing_api_batches
+      response = do_request(NIL_UUID)
+      results = response["results"]
+
+      # todo: can we get this for real? needs a new endpoint... probably not worth it. just supply a different progress reporter?
```

Comment:
I don't think there's a need for a new endpoint; we can just run a separate count query as part of the pagination on the publishing api side. It can change as part of https://trello.com/c/3qIYFDBI/634-finish-publishing-api-endpoint-for-link-checker-tool

OTOH I'm fine with just hardcoding a guess for now as the progress report is just for giving the user some feedback.